merkle-tree/src/routes/merkle-tree.routes.js

```diff
@@ -21,48 +22,80 @@ const alreadyStarting = {}; // initialises as false
 async function startEventFilter(req, res, next) {
   logger.debug('src/routes/merkle-tree.routes startEventFilter()');
 
-  const { contractName, treeId, contractAddress } = req.body; // contractAddress & treeId are optional parameters. Address can instead be inferred by Timber in many cases.
+  const { contractName, treeId, contractId } = req.body; // contractAddress & treeId are optional parameters. Address can instead be inferred by Timber in many cases.
```

Comment:
For backwards compatibility, could we still have `contractAddress` as an optional input here? Then only use the external endpoint further down if we have a contractId

Reply:
So if `contractAddress` doesn't exist, it will still work with a single zapp.

If you follow the trace: `contractController.instantiateContract()` (line 75) -> `getContractInstanceFromBuildFolder()` -> if `contractId` doesn't exist (line 114), if `contractAddress` is `undefined` (line 118) , it goes to `utilsWeb3.getContractAddress(contractName)`and looks up the address from the Truffle's file

Reply:
I see it would work fine - my comment was more for preventing the extra `utilsWeb3.getContractAddress(contractName)` call when the user can just pass it in if they want to. It doesn't make a huge difference but (imo) better to reduce the stack where we can.

Reply:
I agree with Miranda here.